addon/components/section.js

```diff
@@ -30,15 +30,16 @@ export default Component.extend(PropTypeMixin, {
    */
   init () {
     this._super()
-    this.set('state', Ember.Object.create({
-      expanded: this.get('expanded') || this.get('expandedOnInitialRender')
-    }))
+    this.set('state', Ember.Object.create())
+    if (this.get('expanded') === undefined) {
+      this.set('state.expanded', this.get('expandedOnInitialRender'))
+    }
```

Comment:
Instead of setting the publicly accessible `expanded` we're now setting the private state representation of `expanded`. If the consuming context has set `expanded` directly, then we are assuming the consumer is in control of whether or not the section is expanded. If `expanded` is set, then the `didReceiveAttrs` hook will take care of setting `state.expanded`. The check for undefined is made to limit the number of `set` calls we're making, although it _should_ work fine without it.